middle_end/flambda/types/structures/row_like.rec.ml

```diff
@@ -231,7 +267,14 @@ struct
       let join_case env case1 case2 =
         let index = join_index case1.index case2.index in
         let maps_to = Maps_to.join env case1.maps_to case2.maps_to in
-        { maps_to; index }
+        let meet_env =
```

Comment:
This is mostly my fault, as I assumed that env extensions would only be involved in meets, but I would appreciate if you changed `Typing_env_extension.join` to take a `Join_env.t` instead of a `Meet_env.t`.